services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,74 @@
+import org.openkilda.testing.service.elastic.ElasticService
+import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def tagName = "kilda-floodlight"
+        def result = elasticService.getLogs(new ElasticQueryBuilder().setTags(tagName).setTimeRange(300).build())
+        assert result?.hits?.total > 0
+
+        then: "There should be discovery messages"
+        result.hits.hits.any {hit -> hit.source.message.toLowerCase().contains("push discovery package via")}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        setup: "Request a non-existent flow"
+        def flowId = "nonexistentflowid" + System.currentTimeMillis()
+        try {
+            northbound.getFlow(flowId)
+        } catch (e) {
+        }
+        and: "Validate rules on a switch"
+        def switchId = topologyDefinition.switches[2].dpId
+        northbound.validateSwitchRules(switchId)
+        when: "Northbound, Storm and Topology Engine logs for 5 minutes are retrieved"
+        def nbTag = "kilda-northbound"
+        def stormTag = "storm-worker_log"
+        def tpTag = "kilda-tpe"
+
+        def nbResult = false
+        def stormResult = false
+        def tpResult = false
+        def timeout = 60
+
+        Wrappers.wait(timeout, 10) {
+            def nbLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(nbTag).setTimeRange(timeout * 2).
+                    setLevel("ERROR").build())
+            def stormLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(stormTag).
+                    setTimeRange(timeout * 2).setLevel("ERROR").build())
+            def tpLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(tpTag).setTimeRange(timeout * 2).
+                    setLevel("INFO").build())
+
+            if ([nbLogs, stormLogs, tpLogs].every {it?.hits?.total > 0})
+            {
+                nbResult = nbLogs.hits.hits.any { hit -> hit.source.message.contains(flowId) }
+                stormResult = stormLogs.hits.hits.any { hit -> hit.source.message.contains(flowId) }
+                tpResult = tpLogs.hits.hits.any {hit -> hit.source.message.contains(switchId.toString())}
+            }
+            [nbLogs, stormLogs].every {it.hits.hits.any { hit -> hit.source.message.contains(flowId) }}
+            return nbResult && stormResult && tpResult
+        }
+
+        then: "Northbound and Storm should generate ERROR message about being unable to locate flow"
+        nbResult
+        stormResult
+
+        and: "Topology Engine should post an INFO message with rule validation with the switch dpId"
+        tpResult
```

Comment:
You should understand that these verifications make no sense, since your 'wait' block already succeeded at this point (otherwise it would have thrown WaitException). So these lines make no sense except for documentation purpose. This is a little bit not natural from my point of view, maybe this is personal, but I would expect your `when: "Northbound, Storm and Topology Engine logs for 5 minutes are retrieved"` to be rephrased and changed to a `then/expect:` block since it is actually performing the verification right now, it is not just getting logs.

Reply:
These blocks were initially intended to be part of documentation.
Moved the text to the asserts